ts/src/base/Exchange.ts

```diff
@@ -1781,6 +1790,25 @@ export default class Exchange {
     // ------------------------------------------------------------------------
     // METHODS BELOW THIS LINE ARE TRANSPILED FROM JAVASCRIPT TO PYTHON AND PHP
 
+    streamProduceHook (result, messageHash) {
+        const parts = messageHash.split (':');
+        const targetName = parts[0];
+        const targetNameLower = targetName.toLowerCase ();
+        if (targetNameLower.indexOf ('orderbook') >= 0) {
+            this.streamProduce ('orderbooks', result);
+        } else if (targetNameLower.indexOf ('ticker') >= 0) {
+            this.streamProduce ('tickers', result);
+        } else if (targetNameLower.indexOf ('order') >= 0) {
+            this.streamProduce ('orders', result);
+        } else if (targetNameLower.indexOf ('position') >= 0) {
+            this.streamProduce ('positions', result);
+        } else if (targetNameLower.indexOf ('mytrade') >= 0) {
+            this.streamProduce ('myTrades', result);
+        } else if (targetNameLower.indexOf ('trade') >= 0) {
+            this.streamProduce ('trades', result);
+        }
+    }
+
```

Comment:
from this method, we can easily manage, change, add, remove the streams, and no need to update 100 exchanges on each time